### PR TITLE
refactor: showQr_migration_compose

### LINKED ISSUE #1632: refactor: ShowQr Screen Migration to compose
**Summary:**

ShowQr Screen Migration to compose

### PR DESCRIPTION
**Pull request by AvneetSingh2001**
## Issue Fix
Fixes #1632 

Jira Task: [MW-25](https://mifosforge.jira.com/browse/MW-25)

## Description
showQr Screen migration to compose

[MW-25]: https://mifosforge.jira.com/browse/MW-25?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mifospay/src/main/java/org/mifospay/qr/showQr/presenter/ShowQrPresenter.kt [hunk 1]
```diff
@@ -24,16 +24,16 @@ class ShowQrPresenter @Inject constructor(private val mUsecaseHandler: UseCaseHa
...
+//            object : UseCaseCallback<GenerateQr.ResponseValue?> {
+//                override fun onSuccess(response: GenerateQr.ResponseValue?) {
+//                    if (response != null) {
+//                        mShowQrView.showGeneratedQr(response.bitmap)
+//                    }
+//                }
+//
+//                override fun onError(message: String) {}
+//            })
     }
```
Comment (PratyushSingh07) [inline mifospay/src/main/java/org/mifospay/qr/showQr/presenter/ShowQrPresenter.kt]: remove comments

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]

File: mifospay/src/main/java/org/mifospay/qr/showQr/presenter/ShowQrViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                    _showQrUiState.value = ShowQrUiState.Error
+                }
+            })
+    }
+}
+
+data class RequestQrData(
+    val amount: String = "",
+    val vpaId: String = "",
+    val currency: String = "TZA",
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifospay/qr/showQr/presenter/ShowQrViewModel.kt]: make it usd

File: mifospay/src/main/java/org/mifospay/qr/showQr/ui/SetAmountDialog.kt [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+                    ) {
+                        Text(text = stringResource(id = R.string.confirm))
+                    }
+                }
+            }
+        }
+    }
+}
+
+private fun validateInputs(
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifospay/qr/showQr/ui/SetAmountDialog.kt]: keep a flag value and maintain.

File: mifospay/src/main/java/org/mifospay/qr/showQr/ui/ShowQrActivity.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import org.mifospay.common.Constants
+import org.mifospay.theme.MifosTheme
+
+/**
+ * Created by naman on 8/7/17.
+ */
+@AndroidEntryPoint
+class ShowQrActivity : BaseActivity() {
+
+    private lateinit var qrDataString: String
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifospay/qr/showQr/ui/ShowQrActivity.kt]: Move it to viewmodel

File: mifospay/src/main/java/org/mifospay/qr/showQr/ui/ShowQrScreen.kt [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+    )
+
+    if (amountDialogState) {
+        SetAmountDialog(
+            dismissDialog = { amountDialogState = false },
+            prefilledAmount = amount ?: "",
+            resetAmount = {
+                amountDialogState = false
+                amount = null
+                currency = "TZA"
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifospay/qr/showQr/ui/ShowQrScreen.kt]: USD

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: mifospay/src/main/java/org/mifospay/qr/showQr/presenter/ShowQrPresenter.kt (hunk 1 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/qr/showQr/presenter/ShowQrPresenter.kt]: done

File: mifospay/src/main/java/org/mifospay/qr/showQr/presenter/ShowQrViewModel.kt (hunk 2 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/qr/showQr/presenter/ShowQrViewModel.kt]: done

File: mifospay/src/main/java/org/mifospay/qr/showQr/ui/SetAmountDialog.kt (hunk 3 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/qr/showQr/ui/SetAmountDialog.kt]: done

File: mifospay/src/main/java/org/mifospay/qr/showQr/ui/ShowQrActivity.kt (hunk 4 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/qr/showQr/ui/ShowQrActivity.kt]: done

File: mifospay/src/main/java/org/mifospay/qr/showQr/ui/ShowQrScreen.kt (hunk 5 above)
Comment (AvneetSingh2001) [reply mifospay/src/main/java/org/mifospay/qr/showQr/ui/ShowQrScreen.kt]: done

**Pull request merged**